lms/static/js/ccx/async_grade_book_download.js

```diff
@@ -0,0 +1,137 @@
+var defineJs = window.define || RequireJS.define;
+
+defineJs("async_grade_book_download",
+  [
+    'backbone',
+    'underscore',
+    'gettext'
+  ],
+  function (Backbone, _, gettext) {
+    'use strict';
+    var GradeBookDownloadView = Backbone.View.extend({
+
+      events: {
+        "click a#ccx-calculate-grades-csv" : "asyncPrepareGradeBook"
+      },
+
+      initialize: function() {
+        // set interval to fetch list of files that are ready for download.
+        this.clearMessageBoxes();
+        this.showReadyGradeBooks();
+      },
+
+      render: function () {
+        return this;
```

Comment:
So, your `render` function is a no-op? This looks like an anti-pattern to me. Why aren't you using Backbone models, and rendering from JS templates?